torchdynamo/variables/tensor.py

```diff
@@ -70,6 +70,8 @@ def create(cls, tx, proxy, example_value=None, nnmodule=None, **options):
 
         if isinstance(example_value, torch.Tensor):
             proxy.node.meta["example_value"] = example_value.clone()
+            if example_value.grad is not None:
```

Comment:
There is a hiccup here. PyTorch gives this warning message for the added testcase

~~~
/data/home/anijain/miniconda/envs/pytorch_dev/lib/python3.8/site-packages/torch/_tensor.py:1082: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at  aten/src/ATen/core/TensorBody.h:481.)
~~~

Reply:
A hrm.  Can we use the `warnings` module to suppress this?

Reply:
Done